lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
@phated 
Yeah, I got that there may be cases where the vinyl will have non-symbolic stats as well as a `symlink` property, like your example<sup>*</sup>, and those look like they're being handled properly.

My thinking, poorly explained though it has been, was there may also be cases where the `isDirectory()` will return false even though the target of the symlink being created is actually a directory. For such cases I believe this PR will end up _not_ making a junction on Windows, which strikes me as incorrect?

E.g., just copying a link:
```js
vfs.src('./in/my-link-to-a-dir', { resolveSymlinks: false })
  .pipe(vfs.dest('./out'));
```

Does that make any sense? I guess I should've tried describing this in code earlier. 

*) Of course you were only making a point about `dest` here, but I guess such a case would usually just be piped into `vfs.symlink` without the need for an intermediate stage, right?